src/PrintPreview.php

```diff
@@ -0,0 +1,237 @@
+                });
+
+                document.querySelectorAll('select').forEach(select => {
+                    select.setAttribute('disabled', true);
+                });
+
+                document.querySelectorAll('textarea').forEach(select => {
+                    select.setAttribute('readonly', true);
+                });
+
+                document.querySelectorAll('aside').forEach(aside => {
+                    aside.remove();
+                });
+
+                var tables = document.querySelectorAll('table, .table');
+
+                tables.forEach(function(table) {
+                    var maxColumnCount = 0;
+
+                    table.querySelectorAll('tr').forEach(function(row) {
+                        var columnCount = row.children.length;
+                        maxColumnCount = Math.max(maxColumnCount, columnCount);
+                    });
+
+                    if(maxColumnCount > 3) {
+                        table.querySelectorAll('th').forEach(function(th) {
+                            if(!th.hasAttribute('colspan') || th.getAttribute('colspan') < 3) {
+                                th.classList.add('more-than-three-columns');
+                            }
+                        });
+                    }
+                });
+
+                document.querySelectorAll('.ms-auto.d-inline-flex.align-items-center.d-none.d-md-block.my-2').forEach(showentries => {
+                    showentries.remove();
+                });
+            });
+JS;
+        echo Html::scriptBlock($js);
+
+        return true;
+    }
+
+    /**
+     * Get the form page URL for the current classe
+     *
+     * @param boolean $full  path or relative one
+     **/
+    public static function getFormURL($full = true)
+    {
+        /** @var array $CFG_GLPI */
+        global $CFG_GLPI;
+
+        $dir = ($full ? $CFG_GLPI['root_doc'] : '');
+        return "$dir/front/print_preview.form.php";
+    }
+
+    public static function getPrintableTypeLabel($itemtype, $item)
+    {
+        $instance = new $itemtype();
+        $label = $instance->getTabNameForItem($item);
+        if (is_array($label)) {
+            $label = reset($label);
+        }
+        if (empty($label)) {
+            $label = '<span><i class="' . $itemtype::getIcon() . ' me-2"></i>' . $itemtype::getTypeName(0) . '</span>';
+        }
+        return $label;
+    }
+
+    public static function getPrintableTabs($itemtype, $item_id)
+    {
+        $item = new $itemtype();
+        $item->getFromDB($item_id);
+
+        $tabKeys = array_keys($item->defineTabs());
+        $tabKeys = array_map(function ($value) {
+            $parts = explode('$', $value);
+            return $parts[0];
+        }, $tabKeys);
+        $tabKeys = array_diff($tabKeys, self::getUnprintableTypes());
+
+        foreach ($tabKeys as $tabKey) {
+            $cleanTabs[$tabKey] = self::getPrintableTypeLabel($tabKey, $item);
+        }
+        return $cleanTabs ?? [];
+    }
```

Comment:
the `history` tab should not be printable (IMHO) (the data is already exportable in the dedicated tab) and with a large volume of data we'll certainly have loading problems